File: src/main/java/hdzi/editstarters/buildsystem/BuildSystem.java
package hdzi.editstarters.buildsystem;

import com.intellij.openapi.actionSystem.DataContext;
import hdzi.editstarters.EditStarters;
import hdzi.editstarters.dependency.Dependency;
import hdzi.editstarters.dependency.Points;
import hdzi.editstarters.dependency.StarterInfo;
import lombok.Getter;

import java.util.Collection;
import java.util.List;

/**
 * 构建工具父类
 */
public abstract class BuildSystem implements EditStarters {
    @Getter
    private final DataContext context;

    private final EditStarters editStarters;

    @Getter
    private final List<Dependency> dependencies;

    @Getter
    private final Dependency springbootDependency;

    @Getter
    private final boolean springBootProject;

    protected BuildSystem(DataContext context, List<Dependency> dependencies, EditStarters editStarters) {
        this.context = context;
        this.editStarters = editStarters;
        this.dependencies = dependencies;
        this.springbootDependency = Points.find(this.dependencies, new Dependency("org.springframework.boot", "spring-boot"));
        this.springBootProject = this.springbootDependency != null;
    }

    @Override
    public void removeStarters(Collection<StarterInfo> dependencies) {
        this.editStarters.removeStarters(dependencies);
    }

    @Override
    public void addStarters(Collection<StarterInfo> dependencies) {
        this.editStarters.addStarters(dependencies);
    }
}

File: src/main/kotlin/hdzi/editstarters/buildsystem/BuildSystem.kt
package hdzi.editstarters.buildsystem

import com.intellij.openapi.actionSystem.DataContext
import hdzi.editstarters.EditStarters

/**
 * 构建工具父类
 */
abstract class BuildSystem(
    val context: DataContext,
    editStartersGetter: () -> EditStarters,
    dependenciesGetter: () -> List<BuildDependency>
) : EditStarters by editStartersGetter() {
    val existsDependencyDB: Map<String, BuildDependency> =
        dependenciesGetter().associateBy({ it.point }, { it })

    val springbootDependency =
        existsDependencyDB[BuildDependency("org.springframework.boot", "spring-boot").point]

    /**
     * 判断是否是spring boot项目
     */
    val isSpringBootProject: Boolean = springbootDependency != null

}
